### PR TITLE
MUMPFL-15 && MUMPFL-14 : View/Edit local contact information

### PR DESCRIPTION
**Pull request by timlevett**
# Overview

+ Based on [these mockups](https://docs.google.com/a/wisc.edu/file/d/0BxQTQDkqSSRuejJDbWFZSk5QM2c/edit)
+ Add in `n` addresses
+ Edit existing address
+ Delete address
+ Comment section with countdown
+ stored in key/value store in the uPortal database. `TODO: integrate with middleware`
+ Pretty empty state
+ Good enough error state
+ Form validation with assumptions on maxlengths
+ Updated from `4.1.1.14` of myuw frame to `4.1.1.18`

This is currently working in my-test1 if you want to take a gander. If not there are a bunch of screenshots below.

# Screenshots

### Empty
image

### Error state on initial load
image

#### Gif of the Whole Thing
http://goo.gl/weKtgY

#### Initial
image
+ Red = required (UX : Do we want to note that somewhere?)

#### Add second line 

image
+ You can add up to 5 lines

#### Add max lines

http://goo.gl/yOjSXT
#### Filled out
image

#### Saved
image

### Multiple Addresses
image

+ col-sm-6

### Edit Address 1 of 2

image
+ Editing multiple addresses at the same time is supported
+ If you click save on one address, it will save all addresses and put them in view mode on successful save

#### Error state on save

image

+ This is true for new or edit

### CODE REVIEW

File: my-profile-local-contact-impl/pom.xml [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+      <groupId>org.slf4j</groupId>
+      <artifactId>log4j-over-slf4j</artifactId>
+      <scope>runtime</scope>
+    </dependency>
+    <dependency>
+        <groupId>org.apache.commons</groupId>
+        <artifactId>commons-lang3</artifactId>
+    </dependency>
+  </dependencies>
+</project>
```
Comment (vertein) [inline my-profile-local-contact-impl/pom.xml]: new line

Comment (timlevett) [inline my-profile-local-contact-impl/pom.xml]: done

**Comment by vertein** [comment]
The accessibility thing @jhanstra and I attended talked about throwing tabindex on forms makes things better for people.  This is something that a lot (possibly all) of our current stuff doesn't have.  I only bring it up now.  Something we try now or come around to later?

**Comment by timlevett** [comment]
@vertein  - It sounds kind of interesting, but could get weird. I just went through the tab on my-test and it seems to work fine w/o that attribute. Might be something we could polish on later?

**Comment by vertein** [comment]
I am totally fine with doing it later @timlevett.  Seeing the form on the page made me remember the presentation.  Even though it works fine, I guess there's still some assistance devices that need it.  :+1: for this and :+1: to us thinking about this in the future and investigating if it's actually needed.

File: my-profile-webapp/src/main/java/edu/wisc/my/profile/web/LocalContactInformationController.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+import org.springframework.web.bind.annotation.ResponseBody;
+
+import edu.wisc.my.profile.model.ContactInformation;
+import edu.wisc.my.profile.service.LocalContactInformationService;
+
+@Controller
+@RequestMapping("/localContactInfo")
+public class LocalContactInformationController {
+  
+  protected static final Logger logger = LoggerFactory.getLogger(LocalContactInformationController.class);
```
Comment (apetro) [inline my-profile-webapp/src/main/java/edu/wisc/my/profile/web/LocalContactInformationController.java]: It's a non-final class, so I'd have gone with a 

`protected final Logger logger = LoggerFactory.getLogger(getClass());`

but I recognize that reasonable people differ as to whether that's a best practice.

File: my-profile-webapp/src/main/webapp/partials/local-address.html [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        <!-- App Navigation if Needed -->
+        <div class="inner-nav-container">
+            <ul class="inner-nav">
+                <li ><a href="#/main">My Profile</a></li>
+                <li class="active"><a href="#/local">Local Address</a></li>
+                <li><a href="#/emergencyContact">Emergency Contacts</a></li>
+            </ul>
+        </div>
+        <div ng-controller="LocalContactInformationController as localContactInfoCtrl">
+            <div class="alert alert-danger" role="alert" ng-if='error && error.length > 0'>{{error}}</div>
```
Comment (apetro) [inline my-profile-webapp/src/main/webapp/partials/local-address.html]: My editor (and GitHub) flags the ampersands here.  Are those okay or do they need to be escaped?

File: my-profile-webapp/src/main/java/edu/wisc/my/profile/web/LocalContactInformationController.java (hunk 2 above)
Comment (timlevett) [inline my-profile-webapp/src/main/java/edu/wisc/my/profile/web/LocalContactInformationController.java]: Good observation, changed all i created as part of this PR.

File: my-profile-webapp/src/main/webapp/partials/local-address.html (hunk 3 above)
Comment (timlevett) [inline my-profile-webapp/src/main/webapp/partials/local-address.html]: Its flagging them due to html markup, however this is fancy angular stuff. Nothing to worry over.

**Comment by apetro** [comment]
Excellent work. :+1: . Nails the stories of getting to something demoable for storing and retrieving local contact info, with enough extras to be really evocative and I expect drive some healthy, concrete conversation about how this functionality should build out.  Everything built here also looks readily adaptable and evolvable as this continues to get real, so there's very little wasted effort and throwaway code going on here.  Yay `#agile`.

File: my-profile-webapp/src/main/webapp/partials/local-address.html (hunk 3 above)
Comment (vertein) [inline my-profile-webapp/src/main/webapp/partials/local-address.html]: I've seen this a lot too.  Angular stuff in html.  @Github going crazy.

**Comment by jhanstra** [comment]
Awesome job. :+1:

**Pull request merged**